custom/motioncontrollers/MotionController.java

```diff
@@ -34,6 +42,9 @@
 	 */
 	public MotionController(PIDSensor sensor) {
 		this.sensor = sensor;
+		output = null;
+		timer = new Timer();
+		task = new MotionControllerTask();
```

Comment:
The reason to use a ScheduledThreadPoolExecutor is to prevent concurrent tasks from delaying each other because Timers use a single thread. However, MotionControllers only have one task each so as long as each has its own instance of a Timer each loop will have its own thread.